crates/sui-graphql-rpc/src/types/move_object.rs

```diff
@@ -68,11 +68,16 @@ impl MoveObject {
         })
     }
 
-    async fn as_staked_sui(&self) -> Option<StakedSui> {
+    async fn as_stake(&self) -> Option<Stake> {
         self.native_object.data.try_as_move().and_then(|x| {
             if x.type_().is_staked_sui() {
-                Some(StakedSui {
-                    move_obj: self.clone(),
+                Some(Stake {
+                    active_epoch_id: None,
```

Comment:
This is probably not desired as we don't have resolvers on stake for these fields, so move objects converted to stake will not populate these fields. I think the data already exists on the native Sui Object so we just need to convert and map those fields

Reply:
I'll have a look at it.

Reply:
Bumping: you should be able to derive most of these fields from the stored object, as you have done in db data provider

Reply:
Fixed, except reward. I'll do that in a follow up PR